tsdb/engine/tsm1/wal.go

```diff
@@ -409,6 +411,9 @@ func (l *WAL) writeToLog(entry WALEntry) (int, error) {
 	// limit how many concurrent encodings can be in flight.  Since we can only
 	// write one at a time to disk, a slow disk can cause the allocations below
 	// to increase quickly.  If we're backed up, wait until others have completed.
+	l.limiter.Take()
```

Comment:
Note from @lesam: We'd feel more confident about this if `Take` supported a timeout parameter. We're ok adding that in a follow-up PR since the improvements of the current code seem solid.